### ISSUE TITLE
Allow manifestTransforms to be async

### ISSUE THREAD

**Issue by azizhk** [GoogleChrome/workbox#2189]
**Library Affected**: workbox-webpack-plugin, workbox-build

**Feature Request Description**:
I have some asynchronous tasks that I need to run in the manifestTransform function.
(Reading json file to determine dependency tree of routes that I want to precache.)

Can we add an `await` in front of the transform function.
https://github.com/GoogleChrome/workbox/blob/d7a7cc0ea507eeef1d64d967b0d53c7fe8b91e27/packages/workbox-build/src/lib/transform-manifest.js#L118

**Comment by jeffposnick** [comment]
Yeah, I think that should be okay. I'll get that in for the next prerelease.

**Comment by azizhk** [comment]
I've actually used [patch-package](https://github.com/ds300/patch-package) to a pass the asset source to get the contents of the file generated by webpack to the transform function.

I originally thought I should be waiting for the file to get written, but that still caused issues if files from previous builds were still there.

So I added:
```
source: compilation.assets[asset.name].source.bind(compilation.assets[asset.name])
```
to 
https://github.com/GoogleChrome/workbox/blob/d7a7cc0ea507eeef1d64d967b0d53c7fe8b91e27/packages/workbox-webpack-plugin/src/lib/get-manifest-entries-from-compilation.js#L165-L169

And similarly in 
https://github.com/GoogleChrome/workbox/blob/d7a7cc0ea507eeef1d64d967b0d53c7fe8b91e27/packages/workbox-build/src/lib/transform-manifest.js#L82-L88

Now I'm not sure if this should be part of workbox. Looks hacky. Does not affect the contents of the generated service worker file as JSON.stringify ignores functions.
